backend/src/main/java/fk/prof/backend/aggregator/CpuSamplingAggregationBucket.java

```diff
@@ -0,0 +1,83 @@
+package fk.prof.backend.aggregator;
+
+import fk.prof.backend.exception.AggregationFailure;
+import fk.prof.backend.model.request.RecordedProfileIndexes;
+import fk.prof.aggregation.MethodIdLookup;
+import fk.prof.aggregation.cpusampling.CpuSamplingFrameNode;
+import fk.prof.aggregation.cpusampling.CpuSamplingTraceDetail;
+import recording.Recorder;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CpuSamplingAggregationBucket {
+  private final MethodIdLookup methodIdLookup = new MethodIdLookup();
+  private final ConcurrentHashMap<String, CpuSamplingTraceDetail> traceDetailLookup = new ConcurrentHashMap<>();
+
+  public Set<String> getAvailableTraces() {
+    return traceDetailLookup.keySet();
+  }
+
+  public CpuSamplingTraceDetail getTraceDetail(String traceName) {
+    return traceDetailLookup.get(traceName);
+  }
+
+  /**
+   * Generates a unmodifiable view of method id -> method name lookup
+   * This method should be called once the aggregation window, this map is member of, has expired.
+   * Calling this method while window is in progress and receiving profiles for aggregation will result in incomplete method name lookup being generated
+   *
+   * @return returns methodId -> method name lookup map
+   */
+  public Map<Long, String> getMethodNameLookup() {
+    return this.methodIdLookup.generateReverseLookup();
+  }
+
+  /**
+   * Aggregates stack samples in the bucket. Throws {@link AggregationFailure} if aggregation fails
+   *
+   * @param stackSampleWse
+   */
+  public void aggregate(Recorder.StackSampleWse stackSampleWse, RecordedProfileIndexes indexes)
+      throws AggregationFailure {
+
+    for (Recorder.StackSample stackSample : stackSampleWse.getStackSampleList()) {
+      //NOTE: As of now, only those samples are aggregated which are associated with a trace id
+      if (stackSample.hasTraceId()) {
```

Comment:
This is assuming trace_id will be null for samples not associated with any trace and that we are discarding all samples without a trace. I guess, later on, we discussed that trace id is going to be "other" or some special value in that case? Need confirmation for that.